frontend/src/components/UserForm/UserForm.js

```diff
@@ -0,0 +1,198 @@
+import { useMutate } from "../../hooks/use-mutate"
+import { useRef, useState, useEffect} from "react"
+import { useUser } from "../../hooks/user-hooks"
+import { useParams } from "react-router"
+import { CountryDropdown, RegionDropdown } from "react-country-region-selector";
+
+const UserForm = () => {
+  const formRef = useRef()
+  const { updateUser } = useMutate()
+  const { id } = useParams()
+  const user = useUser({ "id": id })
+  const isAdmin = user?.data?.data?.data.user.admin
+
+  const fetchedCountry = user?.data?.data?.data?.user.country;
+  const fetchedState = user?.data?.data?.data?.user.state;
+  const [country, setCountry] = useState("");
+  const [region, setRegion] = useState("");
+
+  useEffect(() => {
+    if (fetchedCountry) setCountry(fetchedCountry);
+    if (fetchedState) setRegion(fetchedState);
+  }, [fetchedCountry, fetchedState]);
+
+  if (user.isError) return <h1>Something went wrong!</h1>
+  if (user.isLoading) return <h1>Loading...</h1>
```

Comment:
Following the changes to the destructured user. this would need to be changed to
```
  if (status === "error") return <h1>Something went wrong!</h1>
  if (status === "loading") return <h1>Loading...</h1>